src/trackers/uniqueKeysTracker.ts

```diff
@@ -18,9 +24,9 @@ const DEFAULT_CACHE_SIZE = 30000;
  */
 export function uniqueKeysTrackerFactory(
   log: ILogger,
-  senderAdapter: IImpressionSenderAdapter,
-  filterAdapter: IFilterAdapter,
-  cacheSize: number = DEFAULT_CACHE_SIZE,
+  filterAdapter: IFilterAdapter = noopFilterAdapter,
+  cacheSize = DEFAULT_CACHE_SIZE,
+  senderAdapter?: IImpressionSenderAdapter,
```

Comment:
As we already discussed, `senderAdapter` will be either our `impressionsCountsCache`, with a new method `recordUniqueKeys`, or a new `unikeKeysCache` with a `track` method.
And it should me required, not optional.

Reply:
I made it optional so i can test functionality without having developed sender adapter. it will be mandatory and use our `impressionsCountCache` in future commits